Google.Api.Gax.Grpc/CallSettings.cs

```diff
@@ -20,18 +20,39 @@ public sealed class CallSettings
         internal const string RequestParamsHeader = "x-goog-request-params";
         internal const string RequestReasonHeader = "x-goog-request-reason";
 
-        internal static CallSettings CancellationTokenNone { get; } = new CallSettings(default(CancellationToken), null, null, null, null, null, null);
+        internal static CallSettings CancellationTokenNone { get; } = new CallSettings(default(CancellationToken), null, null, null, null, null);
 
         /// <summary>
         /// Constructs an instance with the specified settings.
         /// </summary>
         /// <param name="cancellationToken">Cancellation token that can be used for cancelling the call.</param>
+        /// <param name="expiration"><see cref="Expiration"/> to use, or null for default expiration behavior.</param>
+        /// <param name="retry"><see cref="Retry"/> to use, or null for default retry behavior.</param>
+        /// <param name="headerMutation">Action to modify the headers to send at the beginning of the call.</param>
+        /// <param name="writeOptions"><see cref="global::Grpc.Core.WriteOptions"/> that will be used for the call.</param>
+        /// <param name="propagationToken"><see cref="ContextPropagationToken"/> for propagating settings from a parent call.</param>
+        public CallSettings(
+            CancellationToken? cancellationToken,
+            Expiration expiration,
+            RetrySettings retry,
+            Action<Metadata> headerMutation,
+            WriteOptions writeOptions,
+            ContextPropagationToken propagationToken) : this(cancellationToken, expiration, retry, headerMutation, writeOptions, propagationToken, null, null)
+        {
+        }
+
+        /// <summary>
+        /// Obsolete. Plese use <see cref="CallSettings(System.Threading.CancellationToken?, Expiration, RetrySettings, Action{Metadata}, WriteOptions, ContextPropagationToken)"/> instead.
```

Comment:
```suggestion
        /// Obsolete. Please use <see cref="CallSettings(System.Threading.CancellationToken?, Expiration, RetrySettings, Action{Metadata}, WriteOptions, ContextPropagationToken)"/> instead.
```

Reply:
And possibly we want to add the link into the comment, as per FromCallCredentials?